src/Orchard/UI/Navigation/NavigationHelper.cs

```diff
@@ -124,7 +124,7 @@ public static class NavigationHelper {
                     string appPath = currentRequest.ApplicationPath ?? "/";
                     string requestUrl = currentRequest.Path.StartsWith(appPath) ? currentRequest.Path.Substring(appPath.Length) : currentRequest.Path;
 
-                    string modelUrl = menuItem.Href.Replace("~/", appPath);
+                    string modelUrl = menuItem.Href?.Replace("~/", appPath);
```

Comment:
It's incomplete, the next statement will then throw.

Reply:
Right, next commit should prevent it.